### PR TITLE
Introduce a C function that performs parser consistency checks.

### PR DESCRIPTION
**Pull request by DougGregor**
Export a C function that can be used to perform various consistency
checks for each source file it processes. This will be used in the C++
compiler to improve testing coverage for the new parser.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci please test

File: Sources/SwiftParser/CompilerSupport.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    // Parse the source file
+    let sourceFile = parser.parseSourceFile()
+
+    // Round-trip test.
+    if flags & 0x01 != 0 {
+      var bufferArray = [UInt8](buffer)
+      bufferArray.append(0)
+      if "\(sourceFile)" != String(cString: bufferArray) {
+        debugPrint(
+          "\(String(cString: filename)): error: file failed to round-trip")
```
Comment (ahoppen) [inline Sources/SwiftParser/CompilerSupport.swift:41]: Any reason why this is `debugPrint` and not `print`?

**Review by ahoppen — APPROVED** [review]

File: Sources/SwiftParser/CompilerSupport.swift (hunk 1 above)
Comment (DougGregor) [reply Sources/SwiftParser/CompilerSupport.swift:41]: No, that's kinda silly. I'll fix it to use `print`.

**Comment by DougGregor** [comment]
@swift-ci please test

**Pull request merged**